cards3-template.js

```diff
@@ -48,7 +76,24 @@ makeCard.cardName = function() { //--> string
 
 makeCard.isCard = function(thing) { // --> true,false
     // return true if thing is a valid card instance made by this factory
+    var card = makeCard(thing.id);
 
+    if(card == null)
+    {
+        return false;
+    }
+
+    if(thing.id !== card.id)
+    {
+        return false;
+    }
+
+    if(thing.rank !== card.rank)
+    {
+        return false;
+    }
+
+    return true;
 }
 
```

Comment:
This is super close, but Dan was looking for something more along the lines of the following:

makeCard.isCard = function(thing) { // --> true,false
    // return true if thing is a valid card instance made by this factory
    return thing
            && (typeof thing === 'object') // check for null or primitive
            && (thing.name === makeCard.cardName) // check at least one method
            && ('id' in thing) && makeCard.isValidID(thing.id); //check id
}